godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -281,3 +281,189 @@ macro_rules! impl_float_vector_fns {
         }
     };
 }
+
+macro_rules! impl_from_axis_for_vectornaxis {
+    (
+        $EnumType: ty,
+        $Axis: tt
+    ) => {
+        impl From<$Axis> for $EnumType {
+            fn from(_value: $Axis) -> Self {
+                <$EnumType>::$Axis
+            }
+        }
+    };
+}
+
+macro_rules! impl_vec_index_by_axis {
+    (
+        $VectorType: ty,
+        $AxisType: ty,
+        $OutputType: ty,
+        $AxisIdent: ident
+    ) => {
+        impl std::ops::Index<$AxisType> for $VectorType {
+            type Output = $OutputType;
+
+            fn index(&self, _index: $AxisType) -> &Self::Output {
+                &self.$AxisIdent
+            }
+        }
+        impl std::ops::IndexMut<$AxisType> for $VectorType {
+            fn index_mut(&mut self, _index: $AxisType) -> &mut $OutputType {
+                &mut self.$AxisIdent
+            }
+        }
+    };
+}
+
+macro_rules! impl_vectori_swizzle {
+    (
+        $VectorType: ty,
+        $ComponentType: ty
+    ) => {
+        impl $VectorType {
+            /// Vector swizzling.
+            ///
+            /// Example:
+            /// ```
+            /// use godot::prelude::*;
+            /// let vc2 = Vector2i::new(1, 2);
+            /// /* Vec2i to Vec2i */
```

Comment:
Please use `//` style comments and leave an empty line before to help readability 🙂